addons/mail/static/src/core/web/recipients_input.js

```diff
@@ -0,0 +1,304 @@
+                            };
+                            this.openFormViewToCreateResPartner({ context });
+                        };
+                    }
+                    options.push(createOption);
+                    return options;
+                },
+            },
+        ];
+    }
+
+    /** @returns {Object} */
+    getTagsFromMailThread() {
+        const tags = [];
+        for (const recipient of this.state.thread.suggestedRecipients) {
+            const title = _t("%(partner_name)s <%(partner_email)s>", {
+                partner_name: recipient.name,
+                partner_email: recipient.email,
+            });
+            tags.push({
+                id: recipient.partner_id,
+                canEdit: true,
+                text: recipient.name,
+                title,
+                onClick: () => {
+                    if (recipient.partner_id) {
+                        this.openFormViewToEditPartner({
+                            resId: recipient.partner_id,
+                            title: _t("Edit: %s", recipient.name),
+                            context: {
+                                form_view_ref: "base.view_partner_simple_form"
+                            },
+                        });
+                    }
+                },
+                onDelete: () => {
+                    this.state.thread.suggestedRecipients =
+                        this.state.thread.suggestedRecipients.filter(
+                            (suggestedRecipient) =>
+                                suggestedRecipient.partner_id !== recipient.partner_id
+                        );
+                },
+            });
+        }
+        for (const recipient of this.state.thread.additionalRecipients) {
+            const title = _t("%(partner_name)s <%(partner_email)s>", {
+                partner_name: recipient.name,
+                partner_email: recipient.email,
+            });
+            tags.push({
+                id: recipient.partner_id,
+                canEdit: true,
+                text: recipient.name,
+                title,
+                onClick: () => {
+                    if (recipient.partner_id) {
+                        this.openFormViewToEditPartner({
+                            resId: recipient.partner_id,
+                            title: _t("Edit: %s", recipient.name),
+                            context: {
+                                form_view_ref: "base.view_partner_simple_form"
+                            },
+                        });
+                    }
+                },
+                onDelete: () => {
+                    this.state.thread.additionalRecipients =
+                        this.state.thread.additionalRecipients.filter(
+                            (additionalRecipient) =>
+                                additionalRecipient.partner_id !== recipient.partner_id
+                        );
+                },
+            });
+        }
```

Comment:
you could create an arrow function taking `(recipient, recipientField)` and in `onDelete` use: `this.state.thread[recipientField]` to avoid code duplication